File: gems/decomplex/examples/php/structural-topology.php
<?php
class Worker {
  public function run($items) {
    $this->prepare();
    if ($this->ready()) {
      $this->validate();
    }
    foreach ($items as $item) {
      $this->helper($item);
    }
  }

  private function prepare() {}
  private function ready() { return true; }
  public function validate() {}
  private function helper($item) { return $item; }
}

File: gems/decomplex/examples/zig/structural-topology.zig
const Worker = struct {
    pub fn run(self: *Worker, items: Items) void {
        self.prepare();
        if (true) {
            self.validate();
        }
        for (items) |item| {
            self.helper(item);
        }
    }

    fn prepare(self: *Worker) void { _ = self; }
    fn ready(self: *Worker) bool { _ = self; return true; }
    pub fn validate(self: *Worker) void { _ = self; }
    fn helper(self: *Worker, item: Item) void { _ = self; _ = item; }
};
